models/unicycle_first_order.py

```diff
@@ -0,0 +1,54 @@
+from math import *
+from scipy.integrate import odeint
+from models.agent import *
+
+class RobotUnicycleFirstOrder(Agent):
+
+	def __init__(self, size, velocity, k):
+		self.size = size
+		self.velocity = velocity
+		self.k = k
+
+	def model(self, q, t, u):
+		x, y, theta = q
+		v, w = u
+
+		dxdt = v * cos(theta)
+		dydt = v * sin(theta)
+		dthetadt = w
+		return [dxdt, dydt, dthetadt]
+
+	def controller(self, q, qref, uref):
+		x, y, theta = q
+		xref, yref, thetaref = qref
+		vref, wref = uref
+		k1, k2, k3 = self.k
+
+		xe = cos(theta) * (xref - x) + sin(theta) * (yref - y)
+		ye = -sin(theta) * (xref - x) + cos(theta) * (yref - y)
+		thetae = thetaref - theta
+		v = vref * cos(thetae) + k1 * xe
+		w = wref + vref * (k2 * ye + k3 * sin(thetae))
+
+		return [v, w]
```

Comment:
where does the controller come from?

Reply:
Original implementation. They had just this robot for the paper.